smoke-test-pipeline.yml

```diff
@@ -78,16 +74,78 @@ stages:
         status=$?
         [ $status -eq 0 ] && echo "Test Done!" || (echo "Test had issues" && exit 1)
 
+      #failOnStderr: true
+      displayName: "Run Validation Tests"
+    - task: PublishPipelineArtifact@1
+      inputs:
+        path: $(System.DefaultWorkingDirectory)/spk-env/log.txt
+        artifact: test_logs
+      condition: always()
+  - job: infra_integration_tests
+    pool:
+      vmImage: 'Ubuntu 16.04'
+    steps:
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        buildType: 'specific'
+        project: $(SPK_PROJECT_ID)
+        definition: $(SPK_DEFINITION_ID)
+        buildVersionToDownload: 'latestFromBranch'
+        branchName: 'refs/heads/master'
+        itemPattern: "spk*"
+        artifactName: 'spk_linux_node_12'
+        targetPath: '$(System.DefaultWorkingDirectory)'
+    - bash: |
+        shopt -s expand_aliases
+        echo "Current Dir is $(pwd)"
+        chmod +x spk-linux
+        alias spk="$(pwd)/spk-linux"
+        spk --version
+      displayName: "Verify SPK downloaded and executable"
+    - bash: |
+        echo "FUNC_SCRIPT: $FUNC_SCRIPT"
+        curl $FUNC_SCRIPT >| functions.sh
+        chmod +x ./functions.sh
+        echo "TEST_SCRIPT2: $TEST_SCRIPT2"
+        curl $TEST_SCRIPT2 >| infra-validations.sh
+        chmod +x ./infra-validations.sh
+      displayName: "Download integration test scripts"
+      env:
+        FUNC_SCRIPT: $(FUNC_SCRIPT)
+        TEST_SCRIPT2: $(TEST_SCRIPT2)
+    - bash: |
+        # Install the extension
+        export AZURE_DEVOPS_EXT_PAT=$(AZDO_PAT)
+        az extension add --name azure-devops
+
+        # Login to Azure
+        echo "az login --service-principal --username $(SP_APP_ID) --password $(SP_PASS) --tenant $(SP_TENANT)"
```

Comment:
@yradsmikham Do we need to login twice when we have multiple jobs?  `SPK_LOCATION` `ACCESS_TOKEN_SECRET` and other vars are being set twice. Could we move this to pipeline variables?

Reply:
That's correct. When you have run them in separate jobs in parallel, it essentially spins up another agent. So you will need to download and install SPK, and reset other configurations for the agent in order to run the infra tests.

Reply:
To reduce some of the preparation, it might make sense to create a custom build agent for SPK, so that we do not have to repeat some of these steps.